src/Syncify.py

```diff
@@ -224,9 +248,14 @@ def get_download_list(self, playlist):
                 if cleaned_full_file_name not in directory_list:
                     song_artist = song["Artist"]
                     song_title = song["Title"]
-                    future = executor.submit(self.find_youtube_link, song_artist, song_title)
-                    futures.append((future, cleaned_full_file_name))
-                    self.logger.warning("Searching for Song: " + cleaned_full_file_name)
+                    if song.get('VideoID'):
+                        song_list_to_download.append({"title": cleaned_full_file_name,
+                                                      "link": "https://www.youtube.com/watch?v=" + song['VideoID']})
+                        self.logger.warning("Added Song to Download List: " + cleaned_full_file_name + " : " + song['VideoID'])
+                    else:
+                        future = executor.submit(self.find_youtube_link, song_artist, song_title)
+                        futures.append((future, cleaned_full_file_name))
+                        self.logger.warning("Searching for Song: " + cleaned_full_file_name)

```

Comment:
I wonder if this could be moved into the `if "youtube" in playlist_link:` ?

Reply:
There might be a bit of re-arranging though

Reply:
Moving this to the top if block is resulting in more code duplication. e.g. `cleaned_full_file_name` construction and checking if it is already in the directory will be duplicated. For now, I'm leaving this as-is. We could potentially refactor this method later.